feat(Reactions): add video frames

OCannings opened the pull request:
A new VideoFrame component has been added which masks a video and
renders a decorative frame on top of it.

```
<VideoFrame src="hello.mp4" frame="frame-1" />
```

The videos autoplay and loop.

OCannings committed:
feat(Reactions): add video frames

A new VideoFrame component has been added which masks a video and
renders a decorative frame on top of it.

```
<VideoFrame src="hello.mp4" frame="frame-1" />
```

The videos autoplay and loop.

ljkbennett commented:
:eyes:

OCannings commented:
still WIP - opening for comments

ljkbennett commented on apps/juicebox_web/assets/js/reactions/video-frame.js:
```diff
@@ -0,0 +1,83 @@
+import React, { Component, PropTypes } from 'react';
+const frameAsset = require.context('../../images/frames', false);
```
controversial

OCannings replied:
:D suggestions welcome

ljkbennett replied:
No idea really. Would `import * as frameAssets from '../../images/frames';` not work?

OCannings replied:
I don't think so, but I'll revisit this before merging.

ljkbennett replied:
It's no more standards-compliant than this anyway. I think the only way to do it that would be would be to have a manifest file where you export the frames.

OCannings committed:
feat(Reactions): add frame selector

Video frames can now be selected by the user. The base64 video has been
moved into a `{ frame: <frame>, video: <video> }` object to ensure the
correct frame is always applied to the video.

littleowllabs commented:
Ebert has finished reviewing this Pull Request and has found:

* 5 possible new issues (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

<!-- Ebert auto generated comment -->

littleowllabs commented:
Ebert has finished reviewing this Pull Request and has found:

* 6 possible new issues (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

<!-- Ebert auto generated comment -->

littleowllabs commented:
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

<!-- Ebert auto generated comment -->

littleowllabs commented:
Ebert has finished reviewing this Pull Request and has found:

* 2 fixed issues! 🎉

But beware that this branch is **[1 commit behind the `ItsJuice:master` branch](https://github.com/ItsJuice/juicebox/compare/ItsJuice:feat/reaction-frames...ItsJuice:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

<!-- Ebert auto generated comment -->

OCannings committed:
Merge branch 'master' into feat/reaction-frames

ljkbennett commented:
Ebert has finished reviewing this Pull Request and has found:

* 2 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/ItsJuice/juicebox/pulls/65.

<!-- Ebert auto generated comment -->

ljkbennett commented on apps/juicebox_web/assets/js/reactions/player.js:
```diff
@@ -1,33 +1,18 @@
         src={ this.props.reaction }
-        onEnded={ this.onLoop }>
-      </video>
+        frame="frame-1"
```
hard coded?

ljkbennett commented on apps/juicebox_web/assets/js/reactions/reactions.js:
```diff
@@ -3,19 +3,33 @@ import { connect } from 'react-redux';
+import VideoFrame from './video-frame';
+import FrameSelector from './frame-selector';
+import { FRAMES } from './frames';
+import './styles.scss';
```
I've switched to modular styles so I may need to fix this.

OCannings replied:
That shouldn't be a problem, there isn't much in there.

OCannings committed:
chore(Reactions): delete reaction player component

which is unused
